x/btcstkconsumer/keeper/finality_provider_registry.go

```diff
@@ -0,0 +1,126 @@
+package keeper
+
+import (
+	"context"
+
+	"github.com/cosmos/cosmos-sdk/runtime"
+
+	errorsmod "cosmossdk.io/errors"
+	"cosmossdk.io/store/prefix"
+
+	bbn "github.com/babylonlabs-io/babylon/types"
+	btcstaking "github.com/babylonlabs-io/babylon/x/btcstaking/types"
+	btcstktypes "github.com/babylonlabs-io/babylon/x/btcstkconsumer/types"
+)
+
+// SetConsumerFinalityProvider adds the given finality provider to CZ chains KVStore
+func (k Keeper) SetConsumerFinalityProvider(ctx context.Context, fp *btcstaking.FinalityProvider) {
+	store := k.finalityProviderStore(ctx, fp.ConsumerId)
+	fpBytes := k.cdc.MustMarshal(fp)
+	store.Set(fp.BtcPk.MustMarshal(), fpBytes)
+	// Update the finality provider chain store
+	fpChainStore := k.finalityProviderChainStore(ctx)
+	fpChainStore.Set(fp.BtcPk.MustMarshal(), []byte(fp.ConsumerId))
```

Comment:
We should consider use `collections.map` for this one as it seems we store the `fp_btc_pk => chain ID`
or maybe just get it from the btcstaking?

Reply:
We can do this in a future PR. At the time of implementing this code we haven't started exploring the `collections` package. Created https://github.com/babylonlabs-io/babylon/issues/423 to track this